src/rhsmlib/facts/hwprobe.py

```diff
@@ -782,20 +783,28 @@ def get_network_info(self) -> Dict[str, str]:
                 else:
                     net_info["network.fqdn"] = hostname
 
-            try:
-                info: List[tuple] = socket.getaddrinfo(hostname, None, socket.AF_INET, socket.SOCK_STREAM)
-                ip_list = set([x[4][0] for x in info])
-                net_info["network.ipv4_address"] = ", ".join(ip_list)
-            except Exception as err:
-                log.debug("Error during resolving IPv4 address of hostname: %s, %s" % (hostname, err))
+            ipv4_addresses: Set[str] = set()
+            ipv6_addresses: Set[str] = set()
+            for address_tuple in infolist:
+                address: str = address_tuple[4][0]
+                if address_tuple[0] is socket.AF_INET:
+                    ipv4_addresses.add(address)
+                if address_tuple[0] is socket.AF_INET6:
```

Comment:
minor nit: you can use `elif` here, as this condition cannot be true when the former is (and vice versa)